packages/react-select/Spec.md

```diff
@@ -0,0 +1,243 @@
+
+Default/standard Select:
+
+```tsx
+<label htmlFor="selectID">Choose a color</label>
+<Select id="selectID">
+  <option>Red</option>
+  <option>Green</option>
+  <option>Blue</option>
+</Select>
+```
+
+Select with grouped options:
+
+```tsx
+<label htmlFor="selectID">Choose an animal</label>
+<Select id="selectID">
+  <optgroup label="Land">
+    <option>Cat</option>
+    <option>Dog</option>
+    <option>Horse</option>
+  </optgroup>
+  <optgroup label="Water">
+    <option>Dolphin</option>
+    <option>Seal</option>
+    <option>Shark</option>
+  </optgroup>
+</Select>
+```
+
+Inline Select with appearance and size set:
+
+```tsx
+<label htmlFor="selectID">Choose a color</label>
+<Select id="selectID" appearance="filledDarker" size="small" inline>
+  <option>Red</option>
+  <option>Green</option>
+  <option>Blue</option>
+</Select>
+```
+
+Disabled Select with second option selected:
+
+```tsx
+<label htmlFor="selectID">Choose a color</label>
+<Select id="selectID" disabled>
+  <option>Red</option>
+  <option selected>Green</option>
+  <option>Blue</option>
+</Select>
+```
+
+## Variants
+
+### Layout
+
+- Block (default)
+- Inline
+
+### Size
+
+- Small
+- Medium (default)
+- Large
+
+### Appearance
+
+- Filled darker
+- Filled lighter
+- Outline (default)
+- Transparent
+
+### Unsupported Select variants
+
+#### Multiple Selection
+
+The Select component does not support multi-select, and does not support the native `multiselect` attribute. Multiple selection is instead provided through the Combobox component. This is because the native `<select multiple>` has poor accessibility and general UX, and we do not recommend using it. Additionally, because the options are not styleable, there is little benefit to using a wrapped `<Select multiple>` over using the native element directly.
+
+#### Size attribute
+
+For similar reasons to `multiple`, the native `size` attribute is not supported out of the box. It also has very limited styling support, and therefore also has little benefit over the native `<select size="N">`. The Listbox (TODO: confirm name) component is an alternative to using the `size` attribute on a `<select>`.
+
+## API
+
+From [Select.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-select/src/components/Select/Select.types.ts)
+
+### Slots
+
+In this component, `select` is the primary slot. Since `select` is primary, `root` is a separate explicit slot to customize the wrapper.
+
+```ts
+export type SelectSlots = {
+  /** Root of the component, renders as a `<span>`. */
+  root: IntrinsicShorthandProps<'span'>;
+  /** The actual `<select>` element */
+  select: IntrinsicShorthandProps<'select'>;
+  /** the icon, typically a down arrow */
+  icon: IntrinsicShorthandProps<'span'>;
+};
+```
+
+### Children
+
+Children of the `Select` component are rendered as children of the internal `<select>` element. The only children that are supported in practice are the `<optgroup>` and `<option>` elements.
```

Comment:
Is there plan to add `SelectOption` / `SelectOptionGroup`? If no, can you please add clarification to the spec?